jwst/associations/lib/rules_level2b.py

```diff
@@ -319,6 +319,23 @@ def __init__(self, *args, **kwargs):
             ),
             Constraint_Single_Science(self.has_science),
             Constraint_TSO(),
+            Constraint(
+                [
+                    Constraint([
+                        DMSAttrConstraint(
+                            name='exp_type',
+                            sources=['exp_type'],
+                            value=['nrc_tsgrism'],
+                        ),
+                        DMSAttrConstraint(
+                            name='pupil',
+                            sources=['pupil'],
+                            value=['clear'],
+                        )],
+                    )
+                ],
+                reduce=Constraint.notany
+            )
```

Comment:
Note that this dual-level Constraint structure is necessary in order to force the individual constraints on exp_type and pupil value be treated as a logical AND, i.e. the `reduce` operation will now only kick in if both of the inner constraints are true.